app/src/test/java/com/example/memeitupapp/gridmemetest/gridmodeltest/GridModelTest.kt

```diff
@@ -22,31 +24,38 @@ class GridModelTest {
     @get:Rule
     val instantTaskExecutorRule = InstantTaskExecutorRule()
 
-    private val memeService: MemeService = mock()
+    private val mockedGetMemesUseCase: GetMemesUseCase = mock()
+    private val mockedGetMemesFromDataBaseUseCase: GetMemesFromDataBaseUseCase = mock()
+    private val mockedUpdateMemesDataBaseUseCase: UpdateMemesDataBaseUseCase = mock()
     private lateinit var gridMemesModel: GridMemesContract.Model
 
     @Before
     fun setup() {
-        gridMemesModel = GridMemesModel(memeService)
+        gridMemesModel = GridMemesModel(mockedGetMemesUseCase, mockedGetMemesFromDataBaseUseCase, mockedUpdateMemesDataBaseUseCase)
     }
 
     @Test
-    fun `on get memes for grid successfully`() {
+    fun `on GetMemesUseCase gets data from API, update data base`() {
         val mockedMemesResult: Result.Success<List<Meme>> = mock()
-        whenever(memeService.getMemesFromApi()).thenReturn(mockedMemesResult)
-
-        runBlocking { gridMemesModel.getMemes() }
+        whenever(mockedGetMemesUseCase.invoke()).thenReturn(mockedMemesResult)
 
         assertEquals(mockedMemesResult, gridMemesModel.getMemes())
+
+        verify(mockedGetMemesUseCase).invoke()
+        verify(mockedUpdateMemesDataBaseUseCase).invoke(mockedMemesResult.data)
     }
 
     @Test
-    fun `on get memes for grid with error connection`() {
-        val mockedMemesResult: Result.Failure = mock()
-        whenever(memeService.getMemesFromApi()).thenReturn(mockedMemesResult)
+    fun `on GetMemesUseCase fails, invoke GetMemesFromDataBaseUseCase`() {
+        val mockedMemesResultFailure: Result.Failure = mock()
```

Comment:
you can omit this variable because you don't do any assert with it

Reply:
You are right, deleted!